numba/np/ufunc/deviceufunc.py

```diff
@@ -90,24 +90,25 @@ def _fill_arrays(self):
         Get all arguments in array form
         """
         for i, arg in enumerate(self.args):
-            if isinstance(arg, np.ndarray):
-                self.arrays[i] = arg
-            elif self.is_device_array(arg):
+            if self.is_device_array(arg):
                 self.arrays[i] = self.as_device_array(arg)
             elif isinstance(arg, (int, float, complex, np.number)):
                 # Is scalar
                 self.scalarpos.append(i)
             else:
-                raise TypeError("argument #%d has invalid type of %s" \
-                % (i + 1, type(arg) ))
+                array = np.asarray(arg)
+                if array.dtype == np.object:
+                    fmt = "argument #%d has invalid type of %s"
+                    raise TypeError(fmt % (i + 1, type(arg) ))
```

Comment:
Does any test hit this?

Reply:
I can't find any way to hit this - the check of the argument types against the signatures for which the function is declared always prevents me from getting through to here. I think this can be removed.

Reply:
Now removed.

Reply:
Maybe booleans, signature `'boolean(boolean,boolean)'`? I suspect they ought to be bundled into the scalar check, but would they trigger an invalid fall through here?

Reply:
I still can't trigger it with booleans - presently I stand by its removal in 0e0bbc1. Can you provide an example where this check is necessary?